evalml/pipelines/components/transformers/column_selectors.py

```diff
@@ -149,23 +150,36 @@ class SelectByType(ColumnSelector):
     """{}"""
     needs_fitting = False
 
-    def __init__(self, column_types=None, random_seed=0, **kwargs):
-        parameters = {"column_types": column_types}
+    def __init__(self, column_types=None, exclude=False, random_seed=0, **kwargs):
+        parameters = {"column_types": column_types, "exclude": exclude}
         parameters.update(kwargs)
-        Transformer.__init__(
-            Transformer,
+        super().__init__(
             parameters=parameters,
             component_obj=None,
             random_seed=random_seed,
         )
 
     def _check_input_for_columns(self, X):
-        col_types = self.parameters.get("column_types")
-        if col_types and X.ww.select(col_types).empty:
-            raise ValueError("Columns of type {column_types} not found in input data.")
+        pass
```

Comment:
If this function is not used, why is it necessary? Can we just remove any calls to it?

Reply:
Good catch, this was added before inheriting from `Transformer` instead. Will fix!